lib/Compilation.js

```diff
@@ -400,7 +400,7 @@ class Compilation extends Tapable {
 		const dependencies = new Map();
 
 		const addDependency = dep => {
-			const resourceIdent = dep.getResourceIdentifier();
+			const resourceIdent = typeof dep.getResourceIdentifier === 'function' ? dep.getResourceIdentifier() : null;
```

Comment:
While it fixes the error, I’m not sure that this is the right fix. This case should not happen.